pkg/codegen/java/gen.go

```diff
@@ -2308,38 +2301,46 @@ func GeneratePackage(
 		}
 	}
 
+	// Unless we are generating a local package, we use gradle as the build
+	// system. Note that we do not support generating build files for any other
+	// system than gradle at this time.
+	//
+	// If the legacyBuildFiles is set to true, we will only use gradle if the
+	// BuildFiles field is explicitly set to `gradle`. This is to support the
+	// legacy behavior of `pulumi-java-gen`. Once `pulumi-java-gen` is
+	// deprecated, we can remove the legacyBuildFiles flag, and always use
+	// gradle if `local` is false.
+	useGradle := !local && // Only use gradle if we are not generating a local package.
+		// legacy behavior requires explicit gradle setting
+		(legacyBuildFiles && info.BuildFiles == "gradle") ||
+		// new behavior uses gradle by default, unless explicitly disabled
+		(!legacyBuildFiles && info.BuildFiles != "none")
```

Comment:
I feel like I'd make this an `if`, which although a bit more verbose might be clearer and obviate some of the comment. E.g.:

```go
var useGradle bool
if local {
  // ...
  useGradle = false
} else {
  if legacyBuildFiles {
    // The default for legacy invocations is "none", so we need to see an explicit "gradle" setting
    useGradle = info.BuildFiles == "gradle"
  } else {
    // The default for new invocations is gradle, unless "none" is specified explicitly
    useGradle = info.BuildFiles != "none"
  }
}
```

Reply:
Yeah that's nicer!